services/manifest-repo-export-service/pkg/repository/transformer.go

```diff
@@ -360,7 +360,7 @@ func (c *DeployApplicationVersion) Transform(
 			envLocks, appLocks, teamLocks map[string]Lock
 			err                           error
 		)
-		envLocks, err = state.GetEnvironmentLocksFromDB(ctx, transaction, c.Environment)
+		envLocks, err = state.GetEnvironmentLocks(c.Environment)
```

Comment:
We should not read information that will be used for decision making (i.e. deploying or not based on an env lock). If a user creates deploys an application version and creates a lock in quick succession, meaning that the export service has not yet processed the deploy application transformer, this deployment will be blocked by the export service, in spite of being deployed on the manifest. We should only read from the database information that will be written to the manifest

Fix: we read from the manifest